File: server/router/example/ai_model.go
package example

import (
	"github.com/flipped-aurora/gin-vue-admin/server/api/v1"
	"github.com/flipped-aurora/gin-vue-admin/server/middleware"
	"github.com/gin-gonic/gin"
)

type AiModelRouter struct{}

// InitAiModelRouter 初始化 AI模型 路由信息
func (s *AiModelRouter) InitAiModelRouter(Router *gin.RouterGroup, PublicRouter *gin.RouterGroup) {
	aiModelRouter := Router.Group("aiModel").Use(middleware.OperationRecord())
	aiModelRouterWithoutRecord := Router.Group("aiModel")
	aiModelRouterWithoutAuth := PublicRouter.Group("aiModel")

	var aiModelApi = v1.ApiGroupApp.ExampleApiGroup.AiModelApi
	{
		aiModelRouter.POST("createAiModel", aiModelApi.CreateAiModel)             // 新建AI模型
		aiModelRouter.DELETE("deleteAiModel", aiModelApi.DeleteAiModel)           // 删除AI模型
		aiModelRouter.DELETE("deleteAiModelByIds", aiModelApi.DeleteAiModelByIds) // 批量删除AI模型
		aiModelRouter.PUT("updateAiModel", aiModelApi.UpdateAiModel)              // 更新AI模型
		aiModelRouter.POST("setDefault", aiModelApi.SetDefaultAiModel)            // 设置默认AI模型
	}
	{
		aiModelRouterWithoutRecord.GET("findAiModel", aiModelApi.FindAiModel)       // 根据ID获取AI模型
		aiModelRouterWithoutRecord.GET("getAiModelList", aiModelApi.GetAiModelList) // 获取AI模型列表
		aiModelRouterWithoutRecord.GET("getDefault", aiModelApi.GetDefaultAiModel)  // 获取默认AI模型
		aiModelRouterWithoutRecord.GET("getEnabled", aiModelApi.GetEnabledAiModels) // 获取启用的AI模型列表
	}
	{
		aiModelRouterWithoutAuth.GET("getAiModelPublic", aiModelApi.GetAiModelPublic) // AI模型开放接口
	}
}

File: web/src/plugin/tkhelper/api/aiModel.js
import service from '@/utils/request'

// @Tags AiModel
// @Summary 创建AI模型
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body model.AiModel true "创建AI模型"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"创建成功"}"
// @Router /tkhelper/ai-model/create [post]
export const createAiModel = (data) => {
  return service({
    url: '/tkhelper/ai-model/create',
    method: 'post',
    data
  })
}

// @Tags AiModel
// @Summary 删除AI模型
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body model.AiModel true "删除AI模型"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"删除成功"}"
// @Router /tkhelper/ai-model/delete [delete]
export const deleteAiModel = (params) => {
  return service({
    url: '/tkhelper/ai-model/delete',
    method: 'delete',
    params
  })
}

// @Tags AiModel
// @Summary 批量删除AI模型
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Success 200 {string} string "{"success":true,"data":{},"msg":"批量删除成功"}"
// @Router /tkhelper/ai-model/deleteByIds [delete]
export const deleteAiModelByIds = (params) => {
  return service({
    url: '/tkhelper/ai-model/deleteByIds',
    method: 'delete',
    params
  })
}

// @Tags AiModel
// @Summary 更新AI模型
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body model.AiModel true "更新AI模型"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"更新成功"}"
// @Router /tkhelper/ai-model/update [put]
export const updateAiModel = (data) => {
  return service({
    url: '/tkhelper/ai-model/update',
    method: 'put',
    data
  })
}

// @Tags AiModel
// @Summary 用id查询AI模型
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data query model.AiModel true "用id查询AI模型"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"查询成功"}"
// @Router /tkhelper/ai-model/find [get]
export const findAiModel = (params) => {
  return service({
    url: '/tkhelper/ai-model/find',
    method: 'get',
    params
  })
}

// @Tags AiModel
// @Summary 分页获取AI模型列表
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data query request.PageInfo true "分页获取AI模型列表"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"获取成功"}"
// @Router /tkhelper/ai-model/getAiModelList [get]
export const getAiModelList = (params) => {
  return service({
    url: '/tkhelper/ai-model/getAiModelList',
    method: 'get',
    params
  })
}

// @Tags AiModel
// @Summary 设置默认AI模型
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Success 200 {string} string "{"success":true,"data":{},"msg":"设置成功"}"
// @Router /tkhelper/ai-model/setDefault [put]
export const setDefaultAiModel = (params) => {
  return service({
    url: '/tkhelper/ai-model/setDefault',
    method: 'put',
    params
  })
}

// @Tags AiModel
// @Summary 获取默认AI模型
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Success 200 {string} string "{"success":true,"data":{},"msg":"获取成功"}"
// @Router /tkhelper/ai-model/getDefault [get]
export const getDefaultAiModel = () => {
  return service({
    url: '/tkhelper/ai-model/getDefault',
    method: 'get'
  })
}

// @Tags AiModel
// @Summary 获取启用的AI模型列表
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Success 200 {string} string "{"success":true,"data":{},"msg":"获取成功"}"
// @Router /tkhelper/ai-model/getEnabled [get]
export const getEnabledAiModels = () => {
  return service({
    url: '/tkhelper/ai-model/getEnabled',
    method: 'get'
  })
}